server/src/main/java/com/distkv/dst/server/runtime/workerpool/Worker.java

```diff
@@ -67,6 +76,24 @@ public void run() {
           case STR_PUT: {
             StringProtocol.PutRequest strPutRequest =
                 (StringProtocol.PutRequest) internalRequest.getRequest();
+            if (isMaster) {
+              for (SalverClient client : salverClients) {
+                synchronized (client) {
+                  DstStringService service = client.getStringService();
+                  StringProtocol.PutResponse tempResponse =
+                      service.put(strPutRequest).get();
+                  if (tempResponse.getStatus() == CommonProtocol.Status.OK) {
+                    continue;
+                  } else {
+                    CompletableFuture<StringProtocol.PutResponse> future =
+                        (CompletableFuture<StringProtocol.PutResponse>)
+                            internalRequest.getCompletableFuture();
+                    future.complete(StringProtocol.PutResponse.newBuilder()
+                        .setStatus(CommonProtocol.Status.SYNC_ERROR).build());
```

Comment:
What should we do if the response is NOT_OK?
I think it should be crashed here since we can't tolerate this fault now. Add also you need add a detail comment to indicate that `we will fix this in fault_tolerant`.